libraries/Microsoft.Bot.Builder.LanguageGeneration/EvaluationOptions.cs

```diff
@@ -116,6 +119,16 @@ public EvaluationOptions(IList<string> optionsList)
         /// </value>
         public string Locale { get; set; } = null;
 
+        /// <summary>
+        /// Gets or sets the additional properties.
+        /// </summary>
+        /// <value>
+        /// Additional properties.
+        /// </value>
+#pragma warning disable CA2227 // Collection properties should be read only
+        public IDictionary<string, object> Properties { get; set; } = new Dictionary<string, object>();
+#pragma warning restore CA2227 // Collection properties should be read only
+
```

Comment:
What I don't like about this for the random stuff is I cannot easily change this through doing an action.  If it is just plain memory I can do a SetProperty action.